--- SIPMessage.cpp.orig	2009-09-29 10:07:17.000000000 +0200
+++ SIPMessage.cpp	2009-09-29 11:16:39.000000000 +0200
@@ -381,12 +381,13 @@
 	sdp_message_c_connection_add
         (sdp, 0, strdup("IN"), strdup("IP4"), strdup(local_ip),NULL, NULL);
 
-	sdp_message_m_payload_add(sdp,0,strdup("0"));
 	switch (codec) {
 		case RTPuLaw:
+			sdp_message_m_payload_add(sdp,0,strdup("0"));
 			sdp_message_a_attribute_add(sdp,0,strdup("rtpmap"),strdup("0 PCMU/8000"));
 			break;
 		case RTPGSM610:
+			sdp_message_m_payload_add(sdp,0,strdup("3"));
 			sdp_message_a_attribute_add(sdp,0,strdup("rtpmap"),strdup("3 GSM/8000"));
 			break;
 		default: assert(0);
@@ -665,12 +666,13 @@
 	sdp_message_c_connection_add
         (sdp, 0, strdup("IN"), strdup("IP4"), strdup(local_ip),NULL, NULL);
 
-	sdp_message_m_payload_add(sdp,0,strdup("0"));
 	switch (audio_codec) {
 		case RTPuLaw:
+			sdp_message_m_payload_add(sdp,0,strdup("0"));
 			sdp_message_a_attribute_add(sdp,0,strdup("rtpmap"),strdup("0 PCMU/8000"));
 			break;
 		case RTPGSM610:
+			sdp_message_m_payload_add(sdp,0,strdup("3"));
 			sdp_message_a_attribute_add(sdp,0,strdup("rtpmap"),strdup("3 GSM/8000"));
 			break;
 		default: assert(0);
@@ -1026,7 +1028,17 @@
 	osip_message_set_content_type(request, strdup("application/dtmf-relay"));
 	char message[31];
 	// FIXME -- This should probably come from a config file.
-	sprintf(message,"Signal=%i\nDuration=200",info);
+	// Replacing the Signal 12 and 11 for * and #
+	switch (info) {
+                case 12:
+			snprintf(message,sizeof(message),"Signal=#\nDuration=200");
+                        break;
+                case 11:
+			snprintf(message,sizeof(message),"Signal=*\nDuration=200"); 
+                        break;
+                default: 
+			snprintf(message,sizeof(message),"Signal=%i\nDuration=200",info);
+        };
 	sprintf(temp_buf,"%lu",strlen(message));
 	osip_message_set_content_length(request, strdup(temp_buf));
 
