diff --git a/libs/esl/php/esl_wrap.cpp b/libs/esl/php/esl_wrap.cpp
index b5fd7a7..7dab817 100644
--- a/libs/esl/php/esl_wrap.cpp
+++ b/libs/esl/php/esl_wrap.cpp
@@ -857,7 +857,7 @@ SWIG_ZTS_ConvertResourcePtr(zval *z, swig_type_info *ty, int flags TSRMLS_DC) {
   swig_object_wrapper *value;
   void *p;
   int type;
-  char *type_name;
+  const char *type_name;

   value = (swig_object_wrapper *) zend_list_find(z->value.lval, &type);
   if ( flags && SWIG_POINTER_DISOWN ) {
